test/unit/org/apache/cassandra/db/filter/ColumnFilterTest.java

```diff
@@ -18,53 +18,348 @@
+
+        assertEquals("*/*", filter.toString());
+        assertFetchedQueried(true, true, filter, v1, v2, s1);
+        assertCellFetchedQueried(true, true, filter, v2, path0, path1, path2, path3, path4);
+        assertCellFetchedQueried(true, true, filter, s2, path1);
+        assertCellFetchedQueried(true, false, filter, s2, path0, path2, path3, path4);
+    }
+
+    private void testRoundTrips(ColumnFilter cf)
+    {
+        testRoundTrip(cf, MessagingService.VERSION_30);
+        testRoundTrip(cf, MessagingService.VERSION_3014);
+    }
+
+    private void testRoundTrip(ColumnFilter columnFilter, int version)
+    {
+        try
+        {
+            DataOutputBuffer output = new DataOutputBuffer();
+            serializer.serialize(columnFilter, output, version);
+            Assert.assertEquals(serializer.serializedSize(columnFilter, version), output.position());
+            DataInputPlus input = new DataInputBuffer(output.buffer(), false);
+            ColumnFilter deserialized = serializer.deserialize(input, version, metadata);
+            Assert.assertEquals(deserialized, columnFilter);
+        }
+        catch (IOException e)
+        {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    private static void assertFetchedQueried(boolean expectedFetched,
+                                             boolean expectedQueried,
+                                             ColumnFilter filter,
+                                             ColumnDefinition... columns)
+    {
+        for (ColumnDefinition column : columns)
+        {
+            assertEquals(String.format("Expected fetches(%s) to be %s", column.name, expectedFetched), expectedFetched, filter.includes(column));
+            if (expectedFetched)
+                assertEquals(String.format("Expected fetchedColumnIsQueried(%s) to be %s", column.name, expectedQueried), expectedQueried, !filter.canSkipValue(column));
+        }
+    }
+
+    private static void assertCellFetchedQueried(boolean expectedFetched,
+                                                 boolean expectedQueried,
+                                                 ColumnFilter filter,
+                                                 ColumnDefinition column,
+                                                 CellPath... paths)
+    {
+        ColumnFilter.Tester tester = filter.newTester(column);
+
+        for (CellPath path : paths)
+        {
+            int p = ByteBufferUtil.toInt(path.get(0));
+
+            if (tester != null)
+            {
+                assertEquals(String.format("Expected includesAllColumns || tester.includes(%s:%s) to be %s", column.name, p, expectedFetched), expectedFetched, tester.includes(path));
+                if (expectedFetched)
+                    assertEquals(String.format("Expected tester.includes(%s:%s) to be %s", column.name, p, expectedQueried), expectedQueried, !tester.canSkipValue(path));
```

Comment:
As before, the error messages here are mentioning the methods used in 3.11 and the long lines could be broken.